activesupport/lib/active_support/cache/redis_cache_store.rb

```diff
@@ -486,6 +470,37 @@ def serialize_entries(entries, **options)
           end
         end
 
+        def change_counter(key, amount, options)
+          redis.with do |c|
+            c = c.node_for(key) if c.is_a?(Redis::Distributed)
+
+            if options[:expires_in] && supports_expire_nx?
+              c.pipelined do |pipe|
+                pipe.incrby(key, amount)
```

Comment:
I just thought. If we wish to keep that original behavior, then a way to optimize it for most case is to only check for `ttl` if `.incrby(key, amount) == amount`.

If that condition doesn't match, we know for sure we didn't create the key, so there is no point checking TTL.

Reply:
Oh and I merged the refactor.